src/Data/Binary/FloatCast.hs

```diff
@@ -0,0 +1,54 @@
+
+{-# LANGUAGE FlexibleContexts #-}
+
+-- | This module is a literal copy of
+--   <http://hackage.haskell.org/package/reinterpret-cast-0.1.0/docs/src/Data-ReinterpretCast-Internal-ImplArray.html>.
+--
+--   Implements casting via a 1-elemnt STUArray, as described in
```

Comment:
The CBOR library by Well Typed uses a different method which may be even quicker. This is almost the same so it's probably not much slower.